snapm/manager/_systemd.py

```
@@ -0,0 +1,212 @@
+                unit_props = dbus.Interface(unit, _ORG_FREEDESTOP_DBUS_PROPS)
+                active_state = unit_props.Get(
+                    f"{_SYSTEMD_TOP_OBJECT}.Unit", "ActiveState"
+                )
+                if active_state == "active":
+                    _log_info("%s is active.", unit_name)
+                    return
+            except dbus.DBusException:  # pragma: no cover
+                pass
+            time.sleep(0.1)  # pragma: no cover
+
+        raise SnapmTimerError(f"Failed to activate {unit_name}.")  # pragma: no cover
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)